benefits/enrollment/views.py

```diff
@@ -113,15 +110,19 @@ def dispatch(self, request, *args, **kwargs):
         return response
 
 
-@decorator_from_middleware(EligibleSessionRequired)
-def system_error(request):
+class SystemErrorView(AgencySessionRequiredMixin, FlowSessionRequiredMixin, EligibleSessionRequiredMixin, TemplateView):
     """View handler for an enrollment system error."""
 
-    # overwrite origin so that CTA takes user to agency index
-    agency = session.agency(request)
-    session.update(request, origin=agency.index_url)
+    template_name = TEMPLATE_SYSTEM_ERROR
 
-    return TemplateResponse(request, TEMPLATE_SYSTEM_ERROR)
+    def get(self, request, *args, **kwargs):
+        # overwrite origin so that CTA takes user to agency index
+        agency = self.agency
+        session.update(request, origin=agency.index_url)
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        return self.get(request, *args, **kwargs)
```

Comment:
Can you help me understand why we need a POST handler here? Is it the same story we dealt with the `RetryView` where JS POSTs a form back to this view?

Reply:
Oh sorry, I see you described this in the PR. Maybe an inline comment here would help us in the future.